projects/plugins/crm/src/js/components/automations-admin/components/bulk-workflow-actions/index.tsx

```diff
@@ -1,26 +1,61 @@
 import { DropdownMenu, MenuItem } from '@wordpress/components';
-import { dispatch } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
 import { chevronDown } from '@wordpress/icons';
+import { useMutateAutomationWorkflows } from 'crm/data/hooks/mutations';
 import { store } from 'crm/state/store';
+import { useCallback } from 'react';
 import styles from './styles.module.scss';
 
-export const BulkWorkflowActions: React.FC = () => {
+type BulkWorkflowActionsProps = {
+	refetchWorkflows: () => void;
+};
+
+export const BulkWorkflowActions: React.FC< BulkWorkflowActionsProps > = ( {
+	refetchWorkflows,
+} ) => {
 	const dropdownMenuIcon = (
 		<div className={ styles.icon }>
-			<div className={ styles.text }>Bulk action</div>
+			<div className={ styles.text }>{ __( 'Bulk action', 'zero-bs-crm' ) }</div>
 			{ chevronDown }
 		</div>
 	);
 
-	const onActivateClick = ( onClose: () => void ) => () => {
-		dispatch( store ).activateSelectedWorkflows();
-		onClose();
-	};
+	const selectedWorkflows = useSelect( select => select( store ).getSelectedWorkflows(), [] );
+	const workflows = useSelect( select => select( store ).getWorkflows(), [] );
+	const { mutate: mutateWorkflows } = useMutateAutomationWorkflows();
 
-	const onDeactivateClick = ( onClose: () => void ) => () => {
-		dispatch( store ).deactivateSelectedWorkflows();
-		onClose();
-	};
+	const onActivateClick = useCallback(
+		( onClose: () => void ) => () => {
+			for ( const workflowId of selectedWorkflows ) {
+				const workflow = workflows[ workflowId ];
+				mutateWorkflows(
+					{ ...workflow, active: true },
+					{
+						onSuccess: () => refetchWorkflows(),
```

Comment:
This looks like it would refetch workflows after each individual workflow in a bulk action is updated, but it doesn't due to `react-query` magic.

Reply:
![giphy (32)](https://github.com/Automattic/jetpack/assets/3846700/95044ec9-4b7b-4d0d-9698-a9699365e629)